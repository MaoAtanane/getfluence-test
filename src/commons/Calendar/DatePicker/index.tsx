import React, { useCallback, useState } from "react";
import styles from "./datePicker.module.scss";
import classnames from "classnames";
import Button from "../../Button";
import { formatDate, isSameDay } from "../../../utils/date";

type DatePickerProps = {
  value?: Date;
  onChange?: (date: Date) => void;
};

const DatePicker: React.FC<DatePickerProps> = ({
  value = new Date(),
  onChange,
}) => {
  const [date, setDate] = useState(new Date());

  const handlePrevMonth = useCallback(() => {
    setDate((prevDate) => {
      return new Date(prevDate.getFullYear(), prevDate.getMonth() - 1, 1);
    });
  }, []);

  const handleNextMonth = useCallback(() => {
    setDate((prevDate) => {
      return new Date(prevDate.getFullYear(), prevDate.getMonth() + 1, 1);
    });
  }, []);

  const handleDateSelect = useCallback((selectedDate: Date) => {
    onChange?.(selectedDate);
  }, []);

  const renderWeeks = useCallback(() => {
    const currentMonth = date.getMonth();
    const currentYear = date.getFullYear();
    const firstDayOfMonth = new Date(currentYear, currentMonth, 1);
    const lastDayOfMonth = new Date(currentYear, currentMonth + 1, 0);
    const weeks = [];

    let days = [];
    let dayOfWeek = firstDayOfMonth.getDay();
    let currentDay = firstDayOfMonth;

    while (currentDay <= lastDayOfMonth) {
      for (let i = 0; i < 7; i++) {
        const localeCurrentDay = currentDay;
        if (i < dayOfWeek) {
          days.push(<td key={`empty-${i}`}></td>);
        } else {
          days.push(
            <td
              className={classnames(styles.td, {
                [styles.greyText]: currentDay > lastDayOfMonth,
                [styles.selected]: isSameDay(currentDay, value),
              })}
              key={currentDay.toString()}
              onClick={() => {
                handleDateSelect(localeCurrentDay);
              }}
              data-testid={formatDate(currentDay)}
            >
              {currentDay.getDate()}
            </td>
          );
          currentDay = new Date(
            currentDay.getFullYear(),
            currentDay.getMonth(),
            currentDay.getDate() + 1
          );
        }
      }

      weeks.push(<tr key={currentDay.toString()}>{days}</tr>);
      days = [];
      dayOfWeek = 0;
    }

    return weeks;
  }, [date, value, onChange]);

  return (
    <div className={styles.container}>
      <div className={styles.dateNavAndLabelContainer}>
        <h3 className={styles.title}>
          {date.toLocaleDateString("default", {
            month: "long",
            year: "numeric",
          })}
        </h3>
        <div className={styles.navButtonContainer}>
          <Button
            onClick={handlePrevMonth}
            variant={"text"}
            className={styles.button}
          >
            {"<"}
          </Button>
          <Button
            onClick={handleNextMonth}
            variant={"text"}
            className={styles.button}
          >
            {">"}
          </Button>
        </div>
      </div>

      <table>
        <thead>
          <tr>
            <th className={styles.th}>S</th>
            <th className={styles.th}>M</th>
            <th className={styles.th}>T</th>
            <th className={styles.th}>W</th>
            <th className={styles.th}>T</th>
            <th className={styles.th}>F</th>
            <th className={styles.th}>S</th>
          </tr>
        </thead>
        <tbody>{renderWeeks()}</tbody>
      </table>
    </div>
  );
};

export default DatePicker;
