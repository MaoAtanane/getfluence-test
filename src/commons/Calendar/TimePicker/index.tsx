import React from "react";
import styles from "./timePicker.module.scss";
import classnames from "classnames";

type TimePickerProps = {
  value: { hour: string; period: "AM" | "PM" };
  onChange: (time: { hour: string; period: "AM" | "PM" }) => void;
};
const TimePicker: React.FC<TimePickerProps> = ({ value, onChange }) => {
  return (
    <div className={styles.container}>
      <div className={styles.innerContainer}>
        {[
          "12",
          "01",
          "02",
          "03",
          "04",
          "05",
          "06",
          "07",
          "08",
          "09",
          "10",
          "11",
        ].map((hour) => (
          <div
            key={hour}
            className={classnames(styles.hourContainer, {
              [styles.selected]: value.hour === hour && value.period === "AM",
            })}
            onClick={() => onChange({ hour, period: "AM" })}
            data-testid={`${hour} : 00 AM`}
          >
            <div className={styles.hour}>{hour} : 00 AM</div>
          </div>
        ))}
        {[
          "12",
          "01",
          "02",
          "03",
          "04",
          "05",
          "06",
          "07",
          "08",
          "09",
          "10",
          "11",
        ].map((hour) => (
          <div
            key={hour}
            className={classnames(styles.hourContainer, {
              [styles.selected]: value.hour === hour && value.period === "PM",
            })}
            onClick={() => onChange({ hour, period: "PM" })}
          >
            <div className={styles.hour}>{hour} : 00 PM</div>
          </div>
        ))}
      </div>
    </div>
  );
};
export default TimePicker;
