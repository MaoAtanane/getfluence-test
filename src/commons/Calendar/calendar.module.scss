@import "../../styles/variables";

.container {
  display: flex;
  align-items: center;
  justify-content: center;
}
.datePickerContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-right: $large-spacing;
}
.dateInputContainer {
  display: flex;
  align-items: center;
  margin-bottom: $medium-spacing;
  width: 100%;

  .dateDisplay {
    border-radius: 5px;
    background: $light-grey-color;
    padding: $small-spacing $medium-spacing;
    font-weight: 400;
    margin-left: $medium-spacing;
  }
}
