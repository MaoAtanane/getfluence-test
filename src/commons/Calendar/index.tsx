import React, { useEffect } from "react";
import styles from "./calendar.module.scss";
import DatePicker from "./DatePicker";
import { formatDate } from "../../utils/date";
import TimePicker from "./TimePicker";

type CalendarProps = {
  value: Date;
  onChange: (date: Date) => void;
};
const Calendar: React.FC<CalendarProps> = ({ value, onChange }) => {
  const [date, setDate] = React.useState(value);
  const [time, setTime] = React.useState<{ hour: string; period: "AM" | "PM" }>(
    { hour: "12", period: "AM" }
  );

  useEffect(() => {
    onChange(new Date(`${date.toDateString()} ${time.hour}:00 ${time.period}`));
  }, [date, time]);

  return (
    <div className={styles.container}>
      <div className={styles.datePickerContainer}>
        <div className={styles.dateInputContainer}>
          <h3>Data</h3>
          <span className={styles.dateDisplay}>{formatDate(date)}</span>
        </div>
        <DatePicker value={date} onChange={setDate} />
      </div>
      <div>
        <div className={styles.dateInputContainer}>
          <h3>Time</h3>
          <span className={styles.dateDisplay}>
            {time.hour}:00 {time.period}
          </span>
        </div>
        <TimePicker value={time} onChange={setTime} />
      </div>
    </div>
  );
};

export default Calendar;
