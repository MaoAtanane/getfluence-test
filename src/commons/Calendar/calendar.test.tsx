import React from "react";
import { describe, expect, test, vi } from "vitest";
import { fireEvent, render } from "@testing-library/react";
import Calendar from "./index.js";

describe("Calendar test", () => {
  test("should call handleChange when calendar is clicked", () => {
    const onClick = vi.fn();
    const date = new Date("2023-01-01");
    const { getByTestId } = render(
      <Calendar onChange={onClick} value={date} />
    );
    fireEvent.click(getByTestId("2/2/2023"));
    expect(onClick).toHaveBeenCalledWith(new Date("2023-02-02 12:00 AM"));
  });

  test("should call handleChange when time picker is clicked", () => {
    const onClick = vi.fn();
    const date = new Date("2023-01-01");
    const { getByTestId } = render(
      <Calendar onChange={onClick} value={date} />
    );
    fireEvent.click(getByTestId("08 : 00 AM"));
    expect(onClick).toHaveBeenCalledWith(new Date("2023-01-01 08:00 AM"));
  });

  test("should render correctly", () => {
    const onClick = vi.fn();
    const date = new Date("2021-01-01");
    const contained = render(<Calendar onChange={onClick} value={date} />);
    expect(contained).toMatchSnapshot();
  });
});
