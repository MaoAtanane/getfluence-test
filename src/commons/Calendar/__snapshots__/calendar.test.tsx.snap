// Vitest Snapshot v1

exports[`Calendar test > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_ab995a"
      >
        <div
          class="_datePickerContainer_ab995a"
        >
          <div
            class="_dateInputContainer_ab995a"
          >
            <h3>
              Data
            </h3>
            <span
              class="_dateDisplay_ab995a"
            >
              1/1/2021
            </span>
          </div>
          <div
            class="_container_e67d13"
          >
            <div
              class="_dateNavAndLabelContainer_e67d13"
            >
              <h3
                class="_title_e67d13"
              >
                February 2023
              </h3>
              <div
                class="_navButtonContainer_e67d13"
              >
                <button
                  class="_container_1e4b97 _text_1e4b97 _primary_1e4b97 _medium_1e4b97 _button_e67d13 _onlyText_1e4b97"
                >
                  &lt;
                </button>
                <button
                  class="_container_1e4b97 _text_1e4b97 _primary_1e4b97 _medium_1e4b97 _button_e67d13 _onlyText_1e4b97"
                >
                  &gt;
                </button>
              </div>
            </div>
            <table>
              <thead>
                <tr>
                  <th
                    class="_th_e67d13"
                  >
                    S
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    M
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    T
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    W
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    T
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    F
                  </th>
                  <th
                    class="_th_e67d13"
                  >
                    S
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td />
                  <td />
                  <td />
                  <td
                    class="_td_e67d13"
                    data-testid="1/2/2023"
                  >
                    1
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="2/2/2023"
                  >
                    2
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="3/2/2023"
                  >
                    3
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="4/2/2023"
                  >
                    4
                  </td>
                </tr>
                <tr>
                  <td
                    class="_td_e67d13"
                    data-testid="5/2/2023"
                  >
                    5
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="6/2/2023"
                  >
                    6
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="7/2/2023"
                  >
                    7
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="8/2/2023"
                  >
                    8
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="9/2/2023"
                  >
                    9
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="10/2/2023"
                  >
                    10
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="11/2/2023"
                  >
                    11
                  </td>
                </tr>
                <tr>
                  <td
                    class="_td_e67d13"
                    data-testid="12/2/2023"
                  >
                    12
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="13/2/2023"
                  >
                    13
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="14/2/2023"
                  >
                    14
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="15/2/2023"
                  >
                    15
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="16/2/2023"
                  >
                    16
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="17/2/2023"
                  >
                    17
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="18/2/2023"
                  >
                    18
                  </td>
                </tr>
                <tr>
                  <td
                    class="_td_e67d13"
                    data-testid="19/2/2023"
                  >
                    19
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="20/2/2023"
                  >
                    20
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="21/2/2023"
                  >
                    21
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="22/2/2023"
                  >
                    22
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="23/2/2023"
                  >
                    23
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="24/2/2023"
                  >
                    24
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="25/2/2023"
                  >
                    25
                  </td>
                </tr>
                <tr>
                  <td
                    class="_td_e67d13"
                    data-testid="26/2/2023"
                  >
                    26
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="27/2/2023"
                  >
                    27
                  </td>
                  <td
                    class="_td_e67d13"
                    data-testid="28/2/2023"
                  >
                    28
                  </td>
                  <td
                    class="_td_e67d13 _greyText_e67d13"
                    data-testid="1/3/2023"
                  >
                    1
                  </td>
                  <td
                    class="_td_e67d13 _greyText_e67d13"
                    data-testid="2/3/2023"
                  >
                    2
                  </td>
                  <td
                    class="_td_e67d13 _greyText_e67d13"
                    data-testid="3/3/2023"
                  >
                    3
                  </td>
                  <td
                    class="_td_e67d13 _greyText_e67d13"
                    data-testid="4/3/2023"
                  >
                    4
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div>
          <div
            class="_dateInputContainer_ab995a"
          >
            <h3>
              Time
            </h3>
            <span
              class="_dateDisplay_ab995a"
            >
              12
              :00 
              AM
            </span>
          </div>
          <div
            class="_container_35a602"
          >
            <div
              class="_innerContainer_35a602"
            >
              <div
                class="_hourContainer_35a602 _selected_35a602"
                data-testid="12 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  12
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="01 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  01
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="02 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  02
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="03 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  03
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="04 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  04
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="05 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  05
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="06 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  06
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="07 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  07
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="08 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  08
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="09 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  09
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="10 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  10
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
                data-testid="11 : 00 AM"
              >
                <div
                  class="_hour_35a602"
                >
                  11
                   : 00 AM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  12
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  01
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  02
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  03
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  04
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  05
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  06
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  07
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  08
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  09
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  10
                   : 00 PM
                </div>
              </div>
              <div
                class="_hourContainer_35a602"
              >
                <div
                  class="_hour_35a602"
                >
                  11
                   : 00 PM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_ab995a"
    >
      <div
        class="_datePickerContainer_ab995a"
      >
        <div
          class="_dateInputContainer_ab995a"
        >
          <h3>
            Data
          </h3>
          <span
            class="_dateDisplay_ab995a"
          >
            1/1/2021
          </span>
        </div>
        <div
          class="_container_e67d13"
        >
          <div
            class="_dateNavAndLabelContainer_e67d13"
          >
            <h3
              class="_title_e67d13"
            >
              February 2023
            </h3>
            <div
              class="_navButtonContainer_e67d13"
            >
              <button
                class="_container_1e4b97 _text_1e4b97 _primary_1e4b97 _medium_1e4b97 _button_e67d13 _onlyText_1e4b97"
              >
                &lt;
              </button>
              <button
                class="_container_1e4b97 _text_1e4b97 _primary_1e4b97 _medium_1e4b97 _button_e67d13 _onlyText_1e4b97"
              >
                &gt;
              </button>
            </div>
          </div>
          <table>
            <thead>
              <tr>
                <th
                  class="_th_e67d13"
                >
                  S
                </th>
                <th
                  class="_th_e67d13"
                >
                  M
                </th>
                <th
                  class="_th_e67d13"
                >
                  T
                </th>
                <th
                  class="_th_e67d13"
                >
                  W
                </th>
                <th
                  class="_th_e67d13"
                >
                  T
                </th>
                <th
                  class="_th_e67d13"
                >
                  F
                </th>
                <th
                  class="_th_e67d13"
                >
                  S
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td />
                <td />
                <td />
                <td
                  class="_td_e67d13"
                  data-testid="1/2/2023"
                >
                  1
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="2/2/2023"
                >
                  2
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="3/2/2023"
                >
                  3
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="4/2/2023"
                >
                  4
                </td>
              </tr>
              <tr>
                <td
                  class="_td_e67d13"
                  data-testid="5/2/2023"
                >
                  5
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="6/2/2023"
                >
                  6
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="7/2/2023"
                >
                  7
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="8/2/2023"
                >
                  8
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="9/2/2023"
                >
                  9
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="10/2/2023"
                >
                  10
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="11/2/2023"
                >
                  11
                </td>
              </tr>
              <tr>
                <td
                  class="_td_e67d13"
                  data-testid="12/2/2023"
                >
                  12
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="13/2/2023"
                >
                  13
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="14/2/2023"
                >
                  14
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="15/2/2023"
                >
                  15
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="16/2/2023"
                >
                  16
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="17/2/2023"
                >
                  17
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="18/2/2023"
                >
                  18
                </td>
              </tr>
              <tr>
                <td
                  class="_td_e67d13"
                  data-testid="19/2/2023"
                >
                  19
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="20/2/2023"
                >
                  20
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="21/2/2023"
                >
                  21
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="22/2/2023"
                >
                  22
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="23/2/2023"
                >
                  23
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="24/2/2023"
                >
                  24
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="25/2/2023"
                >
                  25
                </td>
              </tr>
              <tr>
                <td
                  class="_td_e67d13"
                  data-testid="26/2/2023"
                >
                  26
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="27/2/2023"
                >
                  27
                </td>
                <td
                  class="_td_e67d13"
                  data-testid="28/2/2023"
                >
                  28
                </td>
                <td
                  class="_td_e67d13 _greyText_e67d13"
                  data-testid="1/3/2023"
                >
                  1
                </td>
                <td
                  class="_td_e67d13 _greyText_e67d13"
                  data-testid="2/3/2023"
                >
                  2
                </td>
                <td
                  class="_td_e67d13 _greyText_e67d13"
                  data-testid="3/3/2023"
                >
                  3
                </td>
                <td
                  class="_td_e67d13 _greyText_e67d13"
                  data-testid="4/3/2023"
                >
                  4
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div>
        <div
          class="_dateInputContainer_ab995a"
        >
          <h3>
            Time
          </h3>
          <span
            class="_dateDisplay_ab995a"
          >
            12
            :00 
            AM
          </span>
        </div>
        <div
          class="_container_35a602"
        >
          <div
            class="_innerContainer_35a602"
          >
            <div
              class="_hourContainer_35a602 _selected_35a602"
              data-testid="12 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                12
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="01 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                01
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="02 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                02
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="03 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                03
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="04 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                04
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="05 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                05
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="06 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                06
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="07 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                07
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="08 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                08
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="09 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                09
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="10 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                10
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
              data-testid="11 : 00 AM"
            >
              <div
                class="_hour_35a602"
              >
                11
                 : 00 AM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                12
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                01
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                02
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                03
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                04
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                05
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                06
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                07
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                08
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                09
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                10
                 : 00 PM
              </div>
            </div>
            <div
              class="_hourContainer_35a602"
            >
              <div
                class="_hour_35a602"
              >
                11
                 : 00 PM
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
